snuba/subscriptions/scheduler.py

```diff
@@ -344,7 +352,33 @@ def __get_subscriptions(self) -> List[Subscription]:
             (current_time - self.__last_refresh).total_seconds() * 1000.0,
             tags={"partition": str(self.__partition_id)},
         )
-        return self.__subscriptions
+
+        if self.__slice_id is not None:
+            filtered_subscriptions: List[Subscription] = []
+            for subscription in self.__subscriptions:
+                # get the EntitySubscription from the Subscription
+                sub_data = subscription.data
+                entity_sub = sub_data.entity_subscription
+
+                # partition key is only defined for SessionsSubscriptions
+                if isinstance(entity_sub, SessionsSubscription):
+                    partition_key_value = entity_sub.get_partitioning_key()
+
+                    # map the partition key's value to the slice ID
+                    logical_part = map_org_id_to_logical_partition(partition_key_value)
```

Comment:
Seems like here you are assuming the partition key for `entity_sub` is always org_id. Is that always the case? Seems like this is tightly coupled to the implementation of `entity_sub` and  would break if we change the partitioning key. SessionSubscription has an organization attribute, maybe we can use that to make it clear we are getting the org_id? Or maybe `map_org_id_to_logical_partition` can be refactored to be more general i.e `map_patition_key_to_logical_partition`

Reply:
That's a good point, and it makes sense. I think it would be good to change it to `map_partition_key_to_logical_partition` as you are suggesting.